src/czt.jl

```diff
@@ -1,107 +1,305 @@
+    return aw, fft(conv_kernel), wd
+end
+
+# type for planning. The arrays are 1D but oriented
+"""
+    CZTPlan_1D{CT, D} # <: AbstractArray{T,D}
+
+type used for the onedimensional plan of the chirp Z transformation (CZT).
+containing
+# Members:
+    `d`: dimension (only one!) to transform with this plan
+    `pad_value`: the value to pad wrapped data with (zero is already handled by the `wd` term, if wanted). 
+    `pad_ranges` :: tuple of two ranges of invalid positions, which can be replaced by pad values
+    `aw`: factor to multiply input with
+    `fft_fv`: fourier-transform (FFTW) of the convolutio kernel
+    `wd`: factor to multiply the result of the convolution by
+    `fftw_plan`: plan for the forward FFTW of the convolution kernel
+    `ifftw_plan`: plan for the inverse FFTW of the convolution kernel
+"""
+struct CZTPlan_1D{CT, PT, D} # <: AbstractArray{T,D}
+    d :: Int
+    pad_value :: PT
+    pad_ranges :: NTuple{2,UnitRange{Int64}}
+    aw :: Array{CT, D}
+    fft_fv :: Array{CT, D}
+    wd :: Array{CT, D}
+    fftw_plan :: FFTW.cFFTWPlan
+    ifftw_plan :: AbstractFFTs.ScaledPlan
+    # dimension of this transformation
+    # as :: Array{T, D} # not needed since it is just the conjugate of ws
+end
+
+"""
+    CZTPlan_ND{CT, D} # <: AbstractArray{T,D}
+
+type used for the onedimensional plan of the chirp Z transformation (CZT).
+containing
+# Members:
+    `plans`: vector of CZTPlan_1D for each of the directions of the ND array to transform
+"""
+struct CZTPlan_ND{CT, PT, D} # <: AbstractArray{T,D}
+    plans :: Vector{CZTPlan_1D{CT,PT, D}}
+end
+
+function get_invalid_ranges(sz, scaled, dsize, dst_center)
+    start_range = 1:0
+    stop_range = 1:0
+    if (scaled*sz < dsize)
+        ceil(Int64, scaled * sz)
+        valid_start = floor(Int, dst_center - (scaled * sz)/2)
+        valid_end = ceil(Int, dst_center + (scaled * sz)/2)
+        start_range = let  
+            if valid_start > 1
+                if valid_end <= dsize
+                    1:valid_start-1
+                else
+                    1+valid_end - dsize:valid_start-1
+                end
+            else
+                1:0
+            end
+        end
+        stop_range = let 
+            if valid_end < dsize
+                if valid_start < 1
+                    valid_end+1:dsize+valid_start-1
+                else
+                    valid_end+1:dsize
+                end
+            else
+                1:0
+            end
+        end
+    end
+    return (start_range, stop_range)
+end
+
+"""
+    plan_czt_1d(xin, scaled, d, dsize=size(xin,d); a=nothing, w=nothing, damp=1.0, src_center=(size(xin,d)+1)/2, dst_center=dsize÷2+1, remove_wrap=false, fft_flags=FFTW.ESTIMATE)
+
+creates a plan for an one-dimensional chirp z-transformation (CZT). The generated plan is then applied via muliplication. For details about the arguments, see czt_1d().
```

Comment:
make reference to other functions correct

Reply:
Not sure what you mean, but I now enclosed this reference and the one in `plan_czt` into single quotation marks.